components/esp32/pm_locks.c

```diff
@@ -174,26 +174,35 @@ esp_err_t esp_pm_dump_locks(FILE* stream)
 
     fprintf(stream, "Lock stats:\n");
     esp_pm_lock_t* it;
+    esp_pm_lock_t lock;
+    char* name;
     SLIST_FOREACH(it, &s_list, next) {
         portENTER_CRITICAL(&it->spinlock);
-        if (it->name == NULL) {
+        lock = *it;
+        name = NULL;
+        if (it->name) {
+            name = strdup(it->name);
```

Comment:
I'm not sure it it is actually necessary to `strdup` the lock name. Documentation for esp_pm_lock_create requires that names are strings with lifetime no less than the lifetime of the lock itself. Since you're holding s_list_lock, the lock is not going to be deleted until the function returns, so using `lock.name` outside of critical section should be fine.

Reply:
Ok nice. Even better. I'll upload a new version.